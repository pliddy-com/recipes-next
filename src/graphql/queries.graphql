# Query for Homepage RecipeCollection (all recipes)

query homePageQuery {
  recipeCollection(order: title_ASC) {
    total
    items {
      ...RecipeSummary
    }
  }
}

# Query for Recipe collection based on filter

query recipeCollection($where: RecipeFilter) {
  recipeCollection(where: $where, order: title_ASC) {
    total
    items {
      ...RecipeSummary
    }
  }
}

# Query for Recipe page

query recipePage($where: RecipeFilter) {
  recipeCollection(where: $where, order: title_ASC, limit: 1) {
    total
    items {
      ...RecipeDefault
    }
  }
}

# Query for List Page RecipeCollection (all recipes matching tag)

query listPageQuery($where: TagFilter) {
  tagCollection(where: $where, limit: 1) {
    total
    items {
      slug
      title
      linkedFrom {
        recipeCollection(limit: 24) {
          total
          items {
            ...RecipeSummary
          }
        }
      }
    }
  }
}

# Query for recipe slugs

query recipeSlugsCollection($where: RecipeFilter) {
  recipeCollection(where: $where) {
    items {
      slug
    }
  }
}

# query for category slugs

query categorySlugsCollection($where: TagFilter) {
  tagCollection(where: $where) {
    items {
      slug
    }
  }
}

# query for category slugs

query tagSlugsCollection($where: TagFilter) {
  tagCollection(where: $where) {
    items {
      slug
      linkedFrom {
        recipeCollection {
          total
        }
      }
    }
  }
}

# query for Taxonomy with Category & Subcategory levels

query taxonomyCollection($where: TaxonomyFilter) {
  taxonomyCollection(where: $where, limit: 1) {
    items {
      ...TaxonomyDefault

      tag {
        ...TagDefault
      }

      childrenCollection(limit: 24) {
        total
        items {
          ... on Tag {
            ...TagDefault

            linkedFrom {
              recipeCollection(limit: 1) {
                total
              }
            }
          }
          ...TaxonomyWithChildren
        }
      }
    }
  }
}
