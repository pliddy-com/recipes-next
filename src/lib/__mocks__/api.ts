export interface ApiTestProps {
  getNavTaxonomy: () => [{ slug: string }];
  getRecipeSlugs: () => string[];
  getTagSlugs: () => string[];
  getRecipeIndex: () => [{ slug: string }];
  getRecipeList: () => [{ slug: string }];
  getRecipePage: () => { slug: string; title: string };
  getTagIndex: () => [{ slug: string; title: string }];
}

// import api library to mock
const api: ApiTestProps = jest.createMockFromModule('lib/api');

api.getNavTaxonomy = jest
  .fn()
  .mockResolvedValue([{ slug: 'slug-1', title: 'Title 1' }]);

api.getRecipeSlugs = jest.fn().mockResolvedValue(['slug-1', 'slug-2']);

api.getTagSlugs = jest.fn().mockResolvedValue(['slug-1', 'slug-2']);

api.getRecipeIndex = jest.fn().mockResolvedValue([
  {
    slug: 'slug-1',
    title: 'Title 1',
  },
  {
    slug: 'slug-2',
    title: 'Title 2',
  },
]);

api.getRecipeList = jest.fn().mockResolvedValue([
  {
    title: 'Recipe List Title',
    slug: 'recipe-list-title',
    linkedFrom: {
      recipeCollection: {
        items: [
          {
            slug: 'slug-1',
            title: 'Title 1',
          },
          {
            slug: 'slug-2',
            title: 'Title 2',
          },
        ],
      },
    },
  },
]);

api.getRecipePage = jest.fn().mockResolvedValue([
  {
    title: 'Recipe Title',
    slug: 'slug-1',
    abstract: 'abstract',
    image: {
      sys: {
        id: 'sysid-1',
        __typename: 'Sys',
      },
      __typename: 'Asset',
      title: 'Image Title',
      description: 'Image description.',
      contentType: 'image/jpeg',
      fileName: 'filename.jpg',
      size: 99999,
      url: 'https://recipes.pliddy.com/url',
      height: 300,
      width: 400,
    },
  },
]);

api.getTagIndex = jest.fn().mockResolvedValue([
  {
    __typename: 'Tag',
    title: 'Tag 1 Title',
    slug: 'tag-1',
    linkedFrom: {
      recipeCollection: {
        total: 2,
        items: [
          {
            __typename: 'Recipe',
            slug: 'slug-1',
            title: 'Title 2',
          },
          {
            __typename: 'Recipe',
            slug: 'slug-2',
            title: 'Title 1',
          },
        ],
      },
    },
  },
  {
    __typename: 'Tag',
    title: 'Tag 2 Title',
    slug: 'tag-2',
    linkedFrom: {
      recipeCollection: {
        total: 2,
        items: [
          {
            __typename: 'Recipe',
            slug: 'slug-3',
            title: 'Title 3',
          },
          {
            __typename: 'Recipe',
            slug: 'slug-4',
            title: 'Title 5',
          },
        ],
      },
    },
  },
]);

module.exports = api;
