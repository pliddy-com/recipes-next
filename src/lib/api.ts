import { queryGraphQLContent } from 'lib/gqlClient';

import {
  NavTaxonomyDocument,
  NavTaxonomyQueryVariables,
  RecipeIndexDocument,
  RecipeIndexQueryVariables,
  RecipeListDocument,
  RecipeListQueryVariables,
  RecipePageDocument,
  RecipePageQueryVariables,
  RecipeSlugsDocument,
  RecipeSlugsQueryVariables,
  TagIndexDocument,
  TagIndexQueryVariables,
  TagSlugsDocument,
  TagSlugsQueryVariables,
} from 'types/queries';

import { hasValue } from 'lib/typeUtils';

// used to query content for nav menu taxonomy on client layout
export const getNavTaxonomy = async (variables: NavTaxonomyQueryVariables) => {
  const { taxonomyCollection } = await queryGraphQLContent(
    NavTaxonomyDocument,
    variables
  );

  return taxonomyCollection ? taxonomyCollection.items.filter(hasValue) : [];
};

// used by getStaticProps for recipe page
export const getRecipeSlugs = async (variables: RecipeSlugsQueryVariables) => {
  const { recipeCollection } = await queryGraphQLContent(
    RecipeSlugsDocument,
    variables
  );

  const results = recipeCollection?.items?.map((child) =>
    child && child.slug ? child.slug : null
  );

  return results ? results?.filter(hasValue) : [];
};

// used by getStaticProps for tag page
export const getTagSlugs = async (variables: TagSlugsQueryVariables) => {
  const { tagCollection } = await queryGraphQLContent(
    TagSlugsDocument,
    variables
  );

  const results = tagCollection?.items
    .filter(hasValue)
    .filter(
      (item) =>
        item?.linkedFrom?.recipeCollection?.total &&
        item?.linkedFrom?.recipeCollection?.total > 0
    )
    .map((child) => (child && child.slug ? child.slug : null));

  return results ? results?.filter(hasValue) : [];
};

// used to query content for home index page
export const getRecipeIndex = async (variables?: RecipeIndexQueryVariables) => {
  const { recipeCollection } = await queryGraphQLContent(
    RecipeIndexDocument,
    variables
  );

  return recipeCollection ? recipeCollection.items.filter(hasValue) : [];
};

// used to query content for tag & category pagepage
export const getRecipeList = async (variables: RecipeListQueryVariables) => {
  const { tagCollection } = await queryGraphQLContent(
    RecipeListDocument,
    variables
  );

  return tagCollection ? tagCollection.items.filter(hasValue) : [];
};

// used to query content for standalone recipe page
export const getRecipePage = async (variables?: RecipePageQueryVariables) => {
  const { recipeCollection } = await queryGraphQLContent(
    RecipePageDocument,
    variables
  );

  return recipeCollection ? recipeCollection.items.filter(hasValue) : [];
};

// used to query content for tag index page
export const getTagIndex = async (variables?: TagIndexQueryVariables) => {
  const { tagCollection } = await queryGraphQLContent(
    TagIndexDocument,
    variables
  );

  const results = tagCollection?.items
    // .filter(hasValue)
    .filter(
      (item) =>
        item?.linkedFrom?.recipeCollection?.total &&
        item?.linkedFrom?.recipeCollection?.total > 0
    );
  // .map((child) => (child && child.slug ? child.slug : null));

  return results ? results?.filter(hasValue) : [];
};
