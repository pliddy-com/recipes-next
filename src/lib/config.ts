interface PageProps {
  defaultDescription?: string;
  defaultTitle?: string;
  description?: string;
  domain?: string;
  title?: string;
}

interface MicrocopyProps {
  category?: PageProps;
  index?: PageProps;
  recipe?: PageProps;
  site?: PageProps;
  tag?: PageProps;
}

export interface ConfigProps {
  microcopy?: MicrocopyProps;
}

const microcopy: MicrocopyProps = {
  category: {
    defaultDescription: 'A collection of recipies categorized as',
    defaultTitle: 'Category',
  },
  index: {
    defaultTitle: 'All Recipes',
    description: "Patrick's recipe collection",
  },
  recipe: {
    defaultDescription: 'A recipe for',
    defaultTitle: 'Recipe',
  },
  site: {
    title: "Patrick's Recipes",
    domain: 'https://recipes.pliddy.com',
  },
  tag: {
    defaultDescription: 'A collection of recipies tagged as',
    defaultTitle: 'Tag',
  },
};

const config: ConfigProps = {
  microcopy,
};

export default config;
