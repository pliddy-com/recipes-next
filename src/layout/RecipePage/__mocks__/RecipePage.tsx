import { RecipeDefaultFragment } from 'types/queries';

jest.createMockFromModule('../RecipePage');

export const RecipePage = jest.fn(
  ({ content }: { content: RecipeDefaultFragment }) => {
    const { title, slug, abstract, image } = content;
    const { fileName } = image ?? {};

    return (
      <div data-testid="RecipePage">
        <h1>{title}</h1>
        <p>{slug}</p>
        <p>{abstract}</p>
        <p>{fileName}</p>
      </div>
    );
  }
);

export default RecipePage;
