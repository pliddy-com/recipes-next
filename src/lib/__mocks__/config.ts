import { ConfigProps } from 'lib/config';

const microcopy = {
  category: {
    defaultDescription: 'Category default description',
    defaultTitle: 'Category Default Title',
  },
  index: {
    defaultTitle: 'Index Default Title',
    description: 'Index default description.',
  },
  recipe: {
    defaultDescription: 'Recipe default description',
    defaultTitle: 'Recipe Default Title',
  },
  site: {
    title: 'Site Title',
  },
  tag: {
    defaultDescription: 'Tag default description',
    defaultTitle: 'Tag Default Title',
  },
};

const config: ConfigProps = {
  microcopy,
};

module.exports = config;
