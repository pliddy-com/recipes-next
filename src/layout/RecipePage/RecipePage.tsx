import Box from '@mui/material/Box';
import Grid from '@mui/material/Grid';
import Stack from '@mui/material/Stack';
import Typography from '@mui/material/Typography';

import DynamicImage from 'components/Image/DynamicImage/DynamicImage';
import EquipmentSection from 'components/RecipeSections/EquipmentSection/EquipmentSection';
import IngredientsSection from 'components/RecipeSections/IngredientsSection/IngredientsSection';
import InstructionsSection from 'components/RecipeSections/InstructionsSection/InstructionsSection';
import NotesSection from 'components/RecipeSections/NotesSection/NotesSection';
import RecipeSchema from 'components/PageHead/Schema/RecipeSchema';
import RichText from 'components/RichText/RichText';
import TagsSection from 'components/RecipeSections/TagsSection/TagsSection';

import { RecipeDefaultFragment, RecipeDescription } from 'types/queries';

import config from './RecipePage.config';
import Container from '@mui/material/Container';

interface RecipePageProps {
  content?: RecipeDefaultFragment;
}

const RecipePage = ({ content }: RecipePageProps) => {
  const {
    title,
    description,
    equipment,
    image,
    ingredientsCollection,
    instructionsCollection,
    notes,
    tagsCollection,
  } = content ?? {};

  const { breakpoints } = config;

  // TODO: pass full description text into rich text component
  //       & destructure json & links inside component (needs types for json & links)
  //       Need to remove typecasting of RecipeDescription

  // const { json, links } = description ?? {};

  const { items: ingredientsSections } = ingredientsCollection ?? {};
  const { items: instructionsSections } = instructionsCollection ?? {};
  const { items: tags } = tagsCollection ?? {};

  return content ? (
    <Container className="page recipe" data-testid="page" maxWidth="xl">
      <RecipeSchema recipe={content} />
      <Box className="recipe" data-testid="RecipePage">
        <Typography variant="h1" gutterBottom>
          {title}
        </Typography>

        <Grid container className="content">
          <Grid item lg={6}>
            <Stack className="description">
              {description && (
                <RichText content={description as RecipeDescription} />
                // <RichText content={description} />
              )}
              {tags && <TagsSection tags={tags} />}
            </Stack>
          </Grid>
          <Grid item lg={6}>
            {' '}
            {image && breakpoints && (
              <Box className="image">
                <DynamicImage
                  image={image}
                  breakpoints={breakpoints}
                  preload={true}
                />
              </Box>
            )}
          </Grid>
        </Grid>

        <Stack direction="column" spacing={3}>
          {ingredientsCollection &&
            ingredientsCollection.items &&
            ingredientsSections && (
              <IngredientsSection sections={ingredientsSections} />
            )}

          {equipment && <EquipmentSection equipment={equipment} />}

          {instructionsSections && (
            <InstructionsSection sections={instructionsSections} />
          )}

          {notes && <NotesSection notes={notes} />}
        </Stack>
      </Box>
    </Container>
  ) : null;
};

export default RecipePage;
