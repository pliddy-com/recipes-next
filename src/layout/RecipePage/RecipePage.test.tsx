// import testing-library methods
import { render } from '@testing-library/react';

// add custom jest matchers from jest-dom
import '@testing-library/jest-dom';

// import the component to test
import RecipePage from './RecipePage';

import { RecipeDefaultFragment } from 'types/queries';

import config from './RecipePage.config';
import * as api from 'lib/api';

jest.mock('lib/api');

jest.createMockFromModule('./RecipePage.config');

describe('Recipe', () => {
  afterEach(() => {
    jest.resetModules();
  });

  describe('when there is page content', () => {
    it('it renders the Recipe', async () => {
      jest.spyOn(api, 'getRecipePage');
      const content = await api.getRecipePage();

      const { asFragment, queryByTestId } = render(
        <RecipePage content={content as RecipeDefaultFragment} />
      );

      // assert that content is rendered
      expect(queryByTestId('RecipePage')).toBeInTheDocument();

      // assert that the component matches the existing snapshot
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('when there is no page content', () => {
    // before each test, delete images node from config
    beforeEach(() => {
      delete config.breakpoints;
    });

    it('it does not render the Recipe', () => {
      const content = undefined;

      render(<RecipePage content={content} />);

      // test if content is not rendered
      render(<RecipePage content={content} />);

      const recipe = document.querySelector('.page');
      expect(recipe).toBeNull();
    });
  });
});
