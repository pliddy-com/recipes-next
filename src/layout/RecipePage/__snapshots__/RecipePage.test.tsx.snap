// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe when there is page content it renders the Recipe 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthXl page recipe css-19r6kue-MuiContainer-root"
    data-testid="page"
  >
    <script
      type="application/ld+json"
    >
      {
    "@context": "https://schema.org/",
    "@type": "Recipe",
    "name": "Recipe Title",
    "image": [
      "https://recipes.pliddy.com/url?w=252&h=252&fit=fill&fm=webp&q=75",
      "https://recipes.pliddy.com/url?w=336&h=252&fit=fill&fm=webp&q=75",
      "https://recipes.pliddy.com/url?w=448&h=252&fit=fill&fm=webp&q=75"
    ],
    "author": {
      "@type": "Person",
      "name": "Patrick Liddy"
    },
    "datePublished": "2022-06-26",
    "description": "Recipe abstract",
    "recipeIngredient": [
      "ingredient 1","ingredient 2"
    ],
    "recipeInstructions": [
      {
        "@type": "HowToSection",
        "name": "Label",
        "itemListElement": [
          {
            "@type": "HowToStep",
            "name": "Label Step 1",
            "text": "Instructions item 1"
          }
        ]
      }
    ]
  }
    </script>
    <div
      class="recipe MuiBox-root css-0"
      data-testid="RecipePage"
    >
      <h1
        class="MuiTypography-root MuiTypography-h1 MuiTypography-gutterBottom css-a4d0r7-MuiTypography-root"
      >
        Recipe Title
      </h1>
      <div
        class="MuiGrid-root MuiGrid-container content css-11lq3yg-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-6 css-1i6hukq-MuiGrid-root"
        >
          <div
            class="description css-nen11g-MuiStack-root"
          >
            <div
              class="MuiBox-root css-0"
              data-testid="richtext"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph css-z2eky3-MuiTypography-root"
              >
                Description rich text
              </p>
            </div>
            <div
              class="tags MuiBox-root css-h5fkc8"
            >
              <div
                class="tagButtons css-m69qwo-MuiStack-root"
                data-testid="tag-buttons"
              >
                <a
                  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall tag css-1k23hlb-MuiButtonBase-root-MuiButton-root"
                  href="/tag/tag-1"
                  role="button"
                  tabindex="0"
                >
                  Tag 1
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-6 css-1i6hukq-MuiGrid-root"
        >
           
          <div
            class="image MuiBox-root css-0"
          >
            <link
              as="image"
              fetchpriority="high"
              href="https://recipes.pliddy.com/url?w=1200&fm=webp&q=75"
              imagesrcset="https://recipes.pliddy.com/url?w=600&fm=webp&q=75 1x, https://recipes.pliddy.com/url?w=1200&fm=webp&q=75 2x"
              media="(min-width: 1048px)"
              rel="preload"
              sizes="600px"
              type="image/webp"
            />
            <link
              as="image"
              fetchpriority="high"
              href="https://recipes.pliddy.com/url?w=1200&fm=webp&q=75"
              imagesrcset="https://recipes.pliddy.com/url?w=500&fm=webp&q=75 1x, https://recipes.pliddy.com/url?w=1000&fm=webp&q=75 2x"
              media="(min-width: 960px) and (max-width: 1047px)"
              rel="preload"
              sizes="500px"
              type="image/webp"
            />
            <link
              as="image"
              fetchpriority="high"
              href="https://recipes.pliddy.com/url?w=1200&fm=webp&q=75"
              imagesrcset="https://recipes.pliddy.com/url?w=900&fm=webp&q=75 1x, https://recipes.pliddy.com/url?w=1800&fm=webp&q=75 2x"
              media="(min-width: 848px) and (max-width: 959px)"
              rel="preload"
              sizes="900px"
              type="image/webp"
            />
            <link
              as="image"
              fetchpriority="high"
              href="https://recipes.pliddy.com/url?w=1200&fm=webp&q=75"
              imagesrcset="https://recipes.pliddy.com/url?w=800&fm=webp&q=75 1x, https://recipes.pliddy.com/url?w=1600&fm=webp&q=75 2x"
              media="(min-width: 748px) and (max-width: 847px)"
              rel="preload"
              sizes="800px"
              type="image/webp"
            />
            <link
              as="image"
              fetchpriority="high"
              href="https://recipes.pliddy.com/url?w=1200&fm=webp&q=75"
              imagesrcset="https://recipes.pliddy.com/url?w=700&fm=webp&q=75 1x, https://recipes.pliddy.com/url?w=1400&fm=webp&q=75 2x"
              media="(min-width: 648px) and (max-width: 747px)"
              rel="preload"
              sizes="700px"
              type="image/webp"
            />
            <link
              as="image"
              fetchpriority="high"
              href="https://recipes.pliddy.com/url?w=1200&fm=webp&q=75"
              imagesrcset="https://recipes.pliddy.com/url?w=600&fm=webp&q=75 1x, https://recipes.pliddy.com/url?w=1200&fm=webp&q=75 2x"
              media="(min-width: 548px) and (max-width: 647px)"
              rel="preload"
              sizes="600px"
              type="image/webp"
            />
            <link
              as="image"
              fetchpriority="high"
              href="https://recipes.pliddy.com/url?w=1200&fm=webp&q=75"
              imagesrcset="https://recipes.pliddy.com/url?w=500&fm=webp&q=75 1x, https://recipes.pliddy.com/url?w=1000&fm=webp&q=75 2x"
              media="(min-width: 448px) and (max-width: 547px)"
              rel="preload"
              sizes="500px"
              type="image/webp"
            />
            <link
              as="image"
              fetchpriority="high"
              href="https://recipes.pliddy.com/url?w=1200&fm=webp&q=75"
              imagesrcset="https://recipes.pliddy.com/url?w=400&fm=webp&q=75 1x, https://recipes.pliddy.com/url?w=800&fm=webp&q=75 2x"
              media="(min-width: 332px) and (max-width: 447px)"
              rel="preload"
              sizes="400px"
              type="image/webp"
            />
            <link
              as="image"
              fetchpriority="high"
              href="https://recipes.pliddy.com/url?w=1200&fm=webp&q=75"
              imagesrcset="https://recipes.pliddy.com/url?w=300&fm=webp&q=75 1x, https://recipes.pliddy.com/url?w=600&fm=webp&q=75 2x"
              media="(max-width: 331px)"
              rel="preload"
              sizes="300px"
              type="image/webp"
            />
            <picture
              style="height: 100%;"
            >
              <source
                media="(min-width: 1048px)"
                sizes="600px"
                srcset="https://recipes.pliddy.com/url?w=600&fm=webp&q=75 1x, https://recipes.pliddy.com/url?w=1200&fm=webp&q=75 2x"
              />
              <source
                media="(min-width: 960px) and (max-width: 1047px)"
                sizes="500px"
                srcset="https://recipes.pliddy.com/url?w=500&fm=webp&q=75 1x, https://recipes.pliddy.com/url?w=1000&fm=webp&q=75 2x"
              />
              <source
                media="(min-width: 848px) and (max-width: 959px)"
                sizes="900px"
                srcset="https://recipes.pliddy.com/url?w=900&fm=webp&q=75 1x, https://recipes.pliddy.com/url?w=1800&fm=webp&q=75 2x"
              />
              <source
                media="(min-width: 748px) and (max-width: 847px)"
                sizes="800px"
                srcset="https://recipes.pliddy.com/url?w=800&fm=webp&q=75 1x, https://recipes.pliddy.com/url?w=1600&fm=webp&q=75 2x"
              />
              <source
                media="(min-width: 648px) and (max-width: 747px)"
                sizes="700px"
                srcset="https://recipes.pliddy.com/url?w=700&fm=webp&q=75 1x, https://recipes.pliddy.com/url?w=1400&fm=webp&q=75 2x"
              />
              <source
                media="(min-width: 548px) and (max-width: 647px)"
                sizes="600px"
                srcset="https://recipes.pliddy.com/url?w=600&fm=webp&q=75 1x, https://recipes.pliddy.com/url?w=1200&fm=webp&q=75 2x"
              />
              <source
                media="(min-width: 448px) and (max-width: 547px)"
                sizes="500px"
                srcset="https://recipes.pliddy.com/url?w=500&fm=webp&q=75 1x, https://recipes.pliddy.com/url?w=1000&fm=webp&q=75 2x"
              />
              <source
                media="(min-width: 332px) and (max-width: 447px)"
                sizes="400px"
                srcset="https://recipes.pliddy.com/url?w=400&fm=webp&q=75 1x, https://recipes.pliddy.com/url?w=800&fm=webp&q=75 2x"
              />
              <source
                media="(max-width: 331px)"
                sizes="300px"
                srcset="https://recipes.pliddy.com/url?w=300&fm=webp&q=75 1x, https://recipes.pliddy.com/url?w=600&fm=webp&q=75 2x"
              />
              <img
                alt="Image description."
                fetchpriority="high"
                height="450"
                loading="eager"
                src="https://recipes.pliddy.com/url?w=1200&fm=webp&q=75"
                style="width: 100%; height: auto; flex-shrink: 0; min-width: 100%; min-height: 100%; object-fit: cover;"
                width="600"
              />
            </picture>
          </div>
        </div>
      </div>
      <div
        class="css-1lwbda4-MuiStack-root"
      >
        <div
          class="ingredients css-1okj3ks-MuiStack-root"
        >
          <h2
            class="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom css-i1hrn3-MuiTypography-root"
          >
            Ingredients
          </h2>
          <div
            class="css-nen11g-MuiStack-root"
          >
            <ul
              class="MuiList-root MuiList-padding recipeList unorderedList css-h4y409-MuiList-root"
            >
              <li
                class="MuiListItem-root MuiListItem-padding css-1tnift2-MuiListItem-root"
              >
                <div
                  class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeMedium css-kw0y8f-MuiSvgIcon-root"
                    data-testid="RestaurantMenuIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m8.1 13.34 2.83-2.83L3.91 3.5c-1.56 1.56-1.56 4.09 0 5.66l4.19 4.18zm6.78-1.81c1.53.71 3.68.21 5.27-1.38 1.91-1.91 2.28-4.65.81-6.12-1.46-1.46-4.2-1.1-6.12.81-1.59 1.59-2.09 3.74-1.38 5.27L3.7 19.87l1.41 1.41L12 14.41l6.88 6.88 1.41-1.41L13.41 13l1.47-1.47z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    ingredient 1
                  </span>
                </div>
              </li>
              <li
                class="MuiListItem-root MuiListItem-padding css-1tnift2-MuiListItem-root"
              >
                <div
                  class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeMedium css-kw0y8f-MuiSvgIcon-root"
                    data-testid="RestaurantMenuIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m8.1 13.34 2.83-2.83L3.91 3.5c-1.56 1.56-1.56 4.09 0 5.66l4.19 4.18zm6.78-1.81c1.53.71 3.68.21 5.27-1.38 1.91-1.91 2.28-4.65.81-6.12-1.46-1.46-4.2-1.1-6.12.81-1.59 1.59-2.09 3.74-1.38 5.27L3.7 19.87l1.41 1.41L12 14.41l6.88 6.88 1.41-1.41L13.41 13l1.47-1.47z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    ingredient 2
                  </span>
                </div>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="equipment css-nen11g-MuiStack-root"
        >
          <h2
            class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
          >
            Equipment
          </h2>
          <ul
            class="MuiList-root MuiList-padding recipeList unorderedList css-h4y409-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-padding css-1tnift2-MuiListItem-root"
            >
              <div
                class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
              >
                <div
                  aria-label="skillet"
                  class="MuiBox-root css-0"
                  data-testid="skillet"
                  role="graphics-symbol"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10.347 8.561c-1.187-1.07-3.385-3.086-4.354-4.19L2.859.59S1.886-.57.945.362L.696.608l-.25.247c-.94.93.207 1.916.207 1.916L4.4 5.946c1.094.982 3.084 3.202 4.141 4.401a8.54 8.54 0 1 0 1.807-1.786ZM2.185 2.75l-.974-.985a.446.446 0 1 1 .634-.626l.974.984a.446.446 0 1 1-.635.627z"
                      style="color: rgb(25, 25, 25);"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  equipment 1
                </span>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-padding css-1tnift2-MuiListItem-root"
            >
              <div
                class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
              >
                <div
                  aria-label="skillet"
                  class="MuiBox-root css-0"
                  data-testid="skillet"
                  role="graphics-symbol"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10.347 8.561c-1.187-1.07-3.385-3.086-4.354-4.19L2.859.59S1.886-.57.945.362L.696.608l-.25.247c-.94.93.207 1.916.207 1.916L4.4 5.946c1.094.982 3.084 3.202 4.141 4.401a8.54 8.54 0 1 0 1.807-1.786ZM2.185 2.75l-.974-.985a.446.446 0 1 1 .634-.626l.974.984a.446.446 0 1 1-.635.627z"
                      style="color: rgb(25, 25, 25);"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  equipment 2
                </span>
              </div>
            </li>
          </ul>
        </div>
        <div
          class="instructions css-1okj3ks-MuiStack-root"
        >
          <h2
            class="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom css-i1hrn3-MuiTypography-root"
          >
            Instructions
          </h2>
          <div
            class="css-nen11g-MuiStack-root"
          >
            <ol
              class="MuiList-root MuiList-padding recipeList orderedList css-h4y409-MuiList-root"
            >
              <li
                class="MuiListItem-root MuiListItem-padding css-1tnift2-MuiListItem-root"
              >
                <div
                  class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
                >
                  <div
                    class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
                  >
                    1.
                  </div>
                </div>
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    Instructions item 1
                  </span>
                </div>
              </li>
            </ol>
          </div>
        </div>
        <div
          class="notes MuiBox-root css-0"
        >
          <h2
            class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
          >
            Notes
          </h2>
          <ul>
            <li>
              note 1
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
